File: Lv. 0/구슬을 나누는 경우의 수.js
function factorial(n) {
  if (n <= 1) return 1;
  return n * factorial(n - 1);
}

const factorial = (n) => (n === 0 ? 1 : n * factorial(n - 1));

function solution(balls, share) {
  var answer = Math.round(
    factorial(balls) / (factorial(balls - share) * factorial(share))
  );
  return answer;
}

File: legacy/programmers_python/Lv. 0/구슬을 나누는 경우의 수.py
## 시간 초과 -> 조합 라이브러리
# from itertools import combinations

# def solution(balls, share):
#     balls_list = [] 
#     for i in range(balls):
#         balls_list.append(i)
#     return len(list(combinations(balls_list,share)))

# math 라이브러리
from math import factorial

def solution(balls, share):
    answer = factorial(balls)/(factorial(balls-share)*factorial(share))
    return answer